### PR TITLE
Start implementing configurable hooks

### PR DESCRIPTION
**Pull request by jameswestman**
This PR, which is still in progress, removes the storefront-info code (it will be moved to a separate repository) and replaces it with a more configurable script-based system. The repository config has a list of hooks (currently just "publish") which are executed in build repositories at certain stages of the pipeline.

It also contains a rewrite of the republishing PR (#79) that works with the new hook-based system.

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -32,6 +33,20 @@ pub struct SubsetConfig {
     pub base_url: Option<String>,
 }
 
+/// A command to run during the build/publish process. Given as an array of arguments, with the first argument being
+/// the path to the program. Arguments are not processed by a shell; they are passed directly to the program.
+#[derive(Deserialize, Debug, Default, Clone)]
+pub struct ConfigHook(Option<Vec<String>>);
```
Comment (danielsilverstone-ct) [inline src/config.rs:39]: I'd be tempted to promote this `Option<>` up to the `ConfigHooks` struct instead.  It feels odd to me for it to be here.

File: src/config.rs [hunk 2]
```diff
@@ -32,6 +33,20 @@ pub struct SubsetConfig {
...
+
+/// Defines a set of hook commands to run at certain points in the build/publish process.
+#[derive(Deserialize, Debug, Default, Clone)]
+#[serde(rename_all = "kebab-case", deny_unknown_fields)]
+pub struct ConfigHooks {
+    /// Runs during publish jobs before the build is imported to the main repository. The hook is allowed to edit the
+    /// repository. The current directory is set to the build directory.
+    pub publish: ConfigHook,
```
Comment (danielsilverstone-ct) [inline src/config.rs:47]: The way this is written, it's valid for the configuration to contain `publish: null` which seems odd/broken.  I'd far rather see this be `pub publish: Option<ConfigHook>` instead.

File: src/app.rs [hunk 3]
```diff
@@ -74,6 +74,10 @@ pub enum ClaimsScope {
     Generate,
     // Permission to list builds and to download a build repo.
     Download,
+    // Permission to republish an app (take it from the repo, re-run the publish hook, and publish it back). Should not
+    // be given to untrusted parties.
```
Comment (danielsilverstone-ct) [inline src/app.rs:78]: I think we need some explanation of what "trusted" means here to guide use.  Maybe that belongs in the code comment, maybe in a README?

**Review by danielsilverstone-ct — COMMENTED** [review]
Thus far, nothing major to pull out.  On this pass I did ignore anything I thought was a copy/paste of code from one place to another (so I entirely ignored the refactor commits) - if you want me to review the code from those, please let me know.

File: src/app.rs (hunk 3 above)
Comment (jameswestman) [reply src/app.rs:78]: Added to the token section in README.md:

> Some token privileges are for managing flat-manager and shouldn't be
given to third parties who are just uploading apps. The token privileges
are described in the ClaimsScope enum in `app.rs`.

**Comment by barthalion** [comment]
@jameswestman Mind resolving conflicts that popped up after merging another PR? Thanks

**Pull request merged**